packages/editor/src/components/color-palette/with-color-context.js

```diff
@@ -12,10 +12,10 @@ import { withSelect } from '@wordpress/data';
 
 export default createHigherOrderComponent(
 	withSelect(
-		( select ) => {
+		( select, ownProps ) => {
 			const settings = select( 'core/editor' ).getEditorSettings();
 			const colors = settings.colors;
-			const disableCustomColors = settings.disableCustomColors;
+			const disableCustomColors = settings.disableCustomColors || ownProps.disableCustomColors;
```

Comment:
In PR https://github.com/WordPress/gutenberg/pull/10457 I'm adding the possibility to overwrite color definitions for a specific component. If we manage the land it before it this PR it may be very useful and simplify the changes here.